src/net_processing.cpp

```diff
@@ -3369,10 +3369,11 @@ void PeerManagerImpl::LogBlockHeader(const CBlockIndex& index, const CNode& peer
     // Having this log by default when not in IBD ensures broad availability of
     // this data in case investigation is merited.
     const auto msg = strprintf(
-        "Saw new %sheader hash=%s height=%d peer=%d%s",
+        "Saw new %sheader%s hash=%s%s peer=%d%s",
         via_compact_block ? "cmpctblock " : "",
-        index.GetBlockHash().ToString(),
-        index.nHeight,
+        via_unsolicited_block ? " via unsolicited block" : "",
```

Comment:
I like the first commit, but the second/third commit are not ideal, because there is a difference between a new block and a new header:
I started up this branch on mainnet while being a few blocks behind (but not that far such that I'd be in IBD so that the log messages would go to `BCLog::VALIDATION`).
Then, the log entries I'd get for each downloaded block ("Saw new header via unsolicited block") are wrong in two ways:
The header is not new (only the block), plus the blocks is not unsolicited (we asked for it with a `getdata` message).

Reply:
That indeed sounds like two bugs, will try to  reproduce.

Reply:
I can reproduce the issue. When the nodes catches up, for a small fraction of blocks it says "Saw new header via unsolicited block". The header refers to a block that was already connected, sometimes more than 25 blocks ago.